src/pyOpenMS/pyopenms/dataframes.py

```diff
@@ -235,33 +235,59 @@ def get_df(self, long : bool = False):
 
 # TODO think about the best way for such top-level function. IMHO in python, encapsulation in a stateless class in unnecessary.
 #   We should probably not just import this whole submodule without prefix.
-def peptide_identifications_to_df(peps: List[PeptideIdentification], decode_ontology : bool = True):
+def peptide_identifications_to_df(peps: List[PeptideIdentification], decode_ontology : bool = True,
+                                  default_missing_values: dict = {bool: False, int: -9999, float: np.nan, str: ''},
+                                  export_unidentified = True):
     """Converts a list of peptide identifications to a pandas DataFrame.
 
     Parameters:
     peps (List[PeptideIdentification]): list of PeptideIdentification objects
     decode_ontology (bool): decode meta value names
+    default_missing_values: default value for missing values for each data type
+    export_unidentified: export PeptideIdentifications without PeptideHit
 
     Returns:
     pandas.DataFrame: peptide identifications in a DataFrame
     """
     switchDict = {bool: '?', int: 'i', float: 'f', str: 'U100'}
+
+    # filter out PeptideIdentifications without PeptideHits if export_unidentified == False
+    if not export_unidentified:
```

Comment:
Thanks for taking my suggestion, however, I would do it during the main iteration.
In C++ this would be fine, but in python this will do copies of each PeptideID.
I make a comment on where I would add it.

Reply:
First I put it in the main iteration (line 302), however, returning None (when there are no peptide hits and export_unidentified == False) does not work with np.fromiter(). Therefore I filtered them out in advance.

Reply:
In this case, I would just "continue" the loop but I think we can leave it for now. Maybe add a TODO.

Reply:
Sorry, I might have gotten it wrong. Yes we would need to think about it.

Reply:
Ok. Is it possible to continue here since we pass only one PeptideIdentification at a time and it is not really a loop?

Reply:
I just checked and yes you can just do "return" without argument.

Reply:
I added a commit

Reply:
Nice!

Reply:
Not sure if the iteration for counting is worth it, but we have to benchmark.